### PR TITLE
Fix minor bugs in Macro

### LINKED ISSUE #157: [PE-D] Macro command should show an example usage if wrong format is entered instead of just "wrong format"
After declaring a macro command, if an input parameter is invalid, the only error message shown is "usage: <macro name> <arguments>
It does not tell the user why their input is invalid.

E.g if a macro parameter is supposed to represent calories and the user enters a negative number for that input, the app doesn't tell the user that only positive numbers are allowed, instead it gives the generic "wrong format" message.

E.g macro testMacro p ; add -n \$ -p \p
testMacro testFood -p -100
The error message shows:
"usage: testMacro
-p <macro argument p>"

Perhaps change the error message to inform the user specifically what input is wrong

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: tjwjoe/ped#5

### LINKED ISSUE #159: [PE-D] Macro command allows user to declare a macro with invalid parameters
For example, I can do `macro testmac ; add -n \$ -p -1000`
Proteins should not accept a negative number, but here it lets me create the macro.

Upon trying to use the macro, the app does not let me press enter and the command is not executed.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: tjwjoe/ped#7

### LINKED ISSUE #162: [PE-D] Inaccurate error message for duplicate fields
Repeated fields (e.g add -n poo -p 50 -p 50) displays "<field> should only contain positive numbers" message. Perhaps consider changing it to something like "Duplicate fields not allowed".

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: tjwjoe/ped#10

### LINKED ISSUE #167: [PE-D] I feel like the Macro command can be explained better in the UG.
Macro is quite an advanced topic and I feel that you could dumb it down further for "average" users to understand more easily. For example, the "FLAG" term is quite abstract to someone who is not familiar with macros. The command format is also a bit difficult to understand, especially for a non-CS user.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: tjwjoe/ped#15

### LINKED ISSUE #187: [PE-D] Expectation of  Flag-less param not at the front differ from documentation
In the documentation, flag-less param is a type of param. Also, in the documentation, param (which includes flag-less param) or optional param can be in any order. However, in the app, I cannot flip the order of flag and flag-less params. If possible, it can be considered to mention that the flag-less argument must be at the front.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: dhafinrazaq/ped#2

### LINKED ISSUE #188: [PE-D] Help macro error
Not sure if this is intended, but maybe it would be good to include "help macro" to see the info of macro

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: dhafinrazaq/ped#3

### LINKED ISSUE #199: Document the behavior of the common-cli library
Many of the bugs reported are related to the fact that common-cli automatically appends arguments of the same parameter
For example, if the user input `add -n Chicken -p 50 -p 50` , the library will parse "50 50" as an argument for protein, which causes parseInt error

### PR DESCRIPTION
**Pull request by JoeyChenSmart**
Close #157, #188, #167, #199, #162, #187, #159 

- [x] Fix macro parse error not showing 'usage' string: #157 
- [x] Fix macro not showing up in `help` command: #188 
- [x] Add message in UG to explain key terms: #167 
- [x] (not macro related but) Add explanation on multiple repeated parameters in UG: #199 #162 maybe #163
- [x] (not macro related but) Add clarification on flag-less parameters in UG: #187
- [x] Explain to users in UG you can create a macro with invalid parameters and error is only thrown when its run: #159

### CODE REVIEW

**Comment by JoeyChenSmart** [comment]
codecov is misbehaving today, can try re-running the CI another time.

Possibly daylight saving problems, considering this post is in the future.

**Review by Jh123x — APPROVED** [review]
LGTM!
Need to wait for CI
